### ISSUE TITLE
macOS Sierra (10.12.2) / Xcode 8.2.1 library compilation errors with Larceny 0.99

### ISSUE THREAD

**Issue by christoff-buerger** [larcenists/larceny#784]
After updating to _macOS Sierra (Version 10.12.2)_ with _Xcode 8.2.1_ it is impossible to compile _R6RS_ libraries or files. The error message of _Larceny 0.99_ is
```
Larceny Panic: Sys/stats.c;677: Assertion failed.
```

I run into this issue with my [_RACR_](https://github.com/christoff-buerger/racr) _Scheme_ library. You can test by executing
```
./install-libraries.bash -s larceny
```

I tried the `compile-stale` script as well as the `compile-stale-libraries` and `compile-library` functions of `(larceny compiler)` with all kind of compiler switches (`slow`, `fast-safe`, `fast-unsafe`, `standard`, `default` and no switch).

Unfortunately, I cannot try the most recent _GitHub_ version as I have no working _Larceny_ compiler anymore (bootstrap for the win)! _Larceny_ itself still runs; "just" compilation of _R6RS_ programs fails.

**Comment by WillClinger** [comment]
Thank you for the detailed bug report.

I tried to update my MacBook to Sierra a while ago, but the update failed.  Now it's urgent.

**Comment by christoff-buerger** [comment]
I tried a bit more. The bug looks kind of random. About 1 out of 20 times a library compiles; the other 19 times is fails with `Larceny Panic: Sys/stats.c;677: Assertion failed`.

I am testing on a _MacBook Air Mid 2011_.

**Comment by WillClinger** [comment]
It looks as though the assertion fails occurs following the first garbage collection in generation 1, when statistics are recorded following the collection.

Sufficiently small and short-running programs can be run by specifying ```-size0 1000M``` on the command line, but that isn't enough for a system build.

I'm going to use an older OS to build a version of Larceny that simply prints a message instead of failing the assertion, and see what I can learn from that.

**Comment by WillClinger** [comment]
Fixed by changeset 2adf75071649c5e840fe47f087d1c006bfe53f25

Evidently macOS Sierra is no longer zeroing out the memory segments allocated using ```malloc``` or similar, which exposed a latent bug in the statistics-gathering code of v0.99.

A prebuilt binary for macOS Sierra should appear on [Larceny's nightly build page](http://www.cesura17.net/%7Elarcenists/Nightly/) by tomorrow morning, and I'll update Larceny's main download page tomorrow as well.

**Comment by christoff-buerger** [comment]
Thanks a lot! I tested it and it works perfectly; all issues are fixed.
